bfgg/controller/actors/dynamodb_resource.py

```diff
@@ -0,0 +1,40 @@
+import boto3
+from datetime import datetime
+
+
+class DynamoDbResource:
+
+    def __init__(self, table: str):
+        dynamo_db = boto3.resource('dynamodb')
+        self.table = dynamo_db.Table(table)
+
+    def save_test_started(self, id: str, start_time: datetime, project: str, test_class: str, java_opts: str = None):
+        item = {
+            'TestId': id,
+            'StartTime': start_time.isoformat(),
+            'Project': project,
+            'TestClass': test_class,
+        }
+        if java_opts:
+            item['JavaOpts'] = java_opts
+        self.table.put_item(Item=item)
+
+    def update_test_ended(self, id: str, end_time: datetime, test_results_url: str):
```

Comment:
this works fine, but its only called if a test successfully finishes - if theres an error or test is manually stopped the end date isnt updated, is this intentional?

Reply:
yeah, you wont get the results and end date if there was an issue with the test, but the details of how the test was run are still available to you. Allows you to rerun them even if they didn't finish, and gives a way to differentiate between 'run' and 'run & completed' tests